addons/common/fnc_removePerFrameHandler.sqf

```diff
@@ -24,24 +24,31 @@ Author:
 
 params [["_handle", -1, [0]]];
 
-if (_handle < 0 || {_handle >= count GVAR(PFHhandles)}) exitWith {};
-
 [{
     params ["_handle"];
 
-    GVAR(perFrameHandlerArray) set [GVAR(PFHhandles) select _handle select 0, {}];
+    private _index = GVAR(PFHhandles) param [_handle];
+    if (isNil "_index") exitWith {false};
 
-    [{
-        params ["_handle"];
+    GVAR(PFHhandles) set [_handle, nil];
+    (GVAR(perFrameHandlerArray) select _index) set [0, {}];
 
-        GVAR(perFrameHandlerArray) deleteAt (GVAR(PFHhandles) select _handle);
-        GVAR(PFHhandles) set [_handle, nil];
+    if (GVAR(perFrameHandlersToRemove) isEqualTo []) then {
```

Comment:
Missing `!` ?

Reply:
Ah never mind I see how it works now, because it's next frame and only runs once due to this